rct229/rulesets/ashrae9012019/section11/section11rule17.py

```diff
@@ -0,0 +1,157 @@
+
+    def __init__(self):
+        super(Section11Rule17, self).__init__(
+            rmds_used=produce_ruleset_model_description(
+                USER=False, BASELINE_0=True, PROPOSED=True
+            ),
+            each_rule=Section11Rule17.RMDRule(),
+            index_rmd=PROPOSED,
+            id="11-6",
+            description="All buildings that will have service water heating loads must include those loads in the simulation.",
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #1, proposed column, (a)",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+            required_fields={"$": ["calendar"], "$.calendar": ["is_leap_year"]},
+            data_items={"is_leap_year": (PROPOSED, "calendar/is_leap_year")},
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section11Rule17.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=False, BASELINE_0=False, PROPOSED=True
+                ),
+                each_rule=Section11Rule17.RMDRule.BuildingSegmentRule(),
+                index_rmd=PROPOSED,
+                list_path="$.buildings[*].building_segments[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_p = context.PROPOSED
+            is_leap_year_p = data["is_leap_year"]
+
+            swh_bat = {
+                bldg_seg_id: get_building_segment_swh_bat(
+                    rmd_p, bldg_seg_id, is_leap_year_p
+                )
+                for bldg_seg_id in find_all(
+                    "$.buildings[*].building_segments[*].id", rmd_p
+                )
+            }
+
+            service_water_heating_use_ids = (
+                get_swh_uses_associated_with_each_building_segment(rmd_p)
+            )
+
+            return {
+                "swh_bat": swh_bat,
+                "service_water_heating_use_ids": service_water_heating_use_ids,
+            }
+
+        class BuildingSegmentRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section11Rule17.RMDRule.BuildingSegmentRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=False, PROPOSED=True
+                    )
+                )
+
+            def create_data(self, context, data):
+                bldg_seg_p = context.PROPOSED
+                bldg_seg_id_p = bldg_seg_p["id"]
+
+                swh_bat = data["swh_bat"]
+                service_water_heating_use_ids = data["service_water_heating_use_ids"]
+
+                return {
+                    "swh_bat": swh_bat[bldg_seg_id_p],
+                    "service_water_heating_use_ids": service_water_heating_use_ids[
+                        bldg_seg_id_p
+                    ],
+                }
```

Comment:
Why do we need a `create_data` in the `RuleDefinitionBase` class?
We could just move the above logic to `get_calc_vals`

Reply:
Good point. Addressed.